Sources/CrowdNotifierSDK/Crypto/CryptoUtils.swift

```diff
@@ -9,22 +9,34 @@
  */
 
 import Foundation
-import CrowdNotifierBaseSDK
 import Clibsodium
 import libmcl
+import HKDF
 
 final class CryptoUtils {
 
     private static let NONCE_LENGTH: Int = 32
 
+    private struct DomainKeys {
+        static let preid = "CN-PREID"
+        static let id = "CN-ID"
+    }
+
     public static func createEncryptedVenueVisits(id: String, arrivalTime: Date, departureTime: Date, venueInfo: VenueInfo) -> [EncryptedVenueVisit] {
         var masterPublicKey = mclBnG2()
-        mclBnG2_deserialize(&masterPublicKey, venueInfo.masterPublicKey.bytes, venueInfo.masterPublicKey.bytes.count)
+        mclBnG2_deserialize(&masterPublicKey, venueInfo.publicKey.bytes, venueInfo.publicKey.bytes.count)
 
         var encryptedVisits = [EncryptedVenueVisit]()
 
         for hour in arrivalTime.hoursSince1970...departureTime.hoursSince1970 {
-            guard let identity = generateIdentity(hour: hour, venueInfo: venueInfo) else {
+            var identityBytes: Bytes?
+            if let qrCodePayload = venueInfo.qrCodePayload {
+                identityBytes = generateIdentityV3(startOfInterval: hour * 3600, qrCodePayload: qrCodePayload.bytes)
```

Comment:
Maybe add small comment here to say that in V3 version, startOfInterval needs to be in seconds since epoch, so need to correct for this.

Reply:
Done.